### PR TITLE
Add hat() and vee() operators to theseus.geometry.so3

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/geometry/functional/test_so3.py [hunk 1]
```diff
@@ -37,4 +41,33 @@ def test_inverse(batch_size: int, dtype: torch.dtype):
     rng.manual_seed(0)
     tangent_vector = torch.rand(batch_size, 3, dtype=dtype, generator=rng)
     group = so3.exp(tangent_vector)
+
+    # check analytic backward for the operator
     check_lie_group_function(so3, "inverse", TEST_EPS, group)
```
Comment (luisenp) [inline tests/geometry/functional/test_so3.py:46]: This is again all duplicated code, can we parameterize these as suggested in the other PR?

File: tests/geometry/functional/test_so3.py [hunk 2]
```diff
@@ -37,4 +41,33 @@ def test_inverse(batch_size: int, dtype: torch.dtype):
...
+
+
+@pytest.mark.parametrize("batch_size", [1, 20, 100])
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+def test_hat(batch_size: int, dtype: torch.dtype):
```
Comment (luisenp) [inline tests/geometry/functional/test_so3.py:51]: More duplicated code.

**Review by luisenp — CHANGES_REQUESTED** [review]
LGTM, but same comment we had earlier about duplicated code.

File: tests/geometry/functional/test_so3.py (hunk 1 above)
Comment (fantaosha) [reply tests/geometry/functional/test_so3.py:46]: I'm planning to add additional tests for each function, not only about analytical backward.

Comment (mhmukadam) [reply tests/geometry/functional/test_so3.py:46]: Should be okay to proceed with reviewing and merging the stream of current PRs, but let's remember to do a pass to resolve any code duplication once all the initial implementation converges.

File: theseus/geometry/functional/so3.py [hunk 3]
```diff
@@ -233,3 +243,89 @@ def backward(cls, ctx, grad_output):
 _jinverse_autograd_fn = _jinverse_impl
 
 inverse, jinverse = lie_group.UnaryOperatorFactory(_module, "inverse")
+
+
+# -----------------------------------------------------------------------------
+# Hat
+# -----------------------------------------------------------------------------
+def _hat_impl(tangent_vector: torch.Tensor) -> torch.Tensor:
```
Comment (mhmukadam) [inline theseus/geometry/functional/so3.py:251]: Do we want to add unit tests for hat/vee functions?

**Review by mhmukadam — COMMENTED** [review]
LGTM!

**Pull request merged**